test/string-class-test.js

```diff
@@ -0,0 +1,172 @@
+  it('should return ANDELA if anDelA.toUpper() was executed', () => {
+    assert.equal('anDelA'.toUpper(), 'ANDELA');
+  });
+  it('should return LOWER CASE for "lower case".toUpper() \
+    was executed', () => {
+    assert.equal('lower case'.toUpper(), 'LOWER CASE');
+  });
+  it('should return 123-456 for 123-456.toUpper()', () => {
+    assert.equal('123-456'.toUpper(), '123-456');
+  });
+});
+
+describe('Test for toLower() method', () => {
+  it('should return "this is" for "THIS IS".toLower()', () => {
+    assert.equal('THIS IS'.toLower(), 'this is');
+  });
+  it('should return "already lower" for "already lower".toLower()', () => {
+    assert.equal('already lower'.toLower(), 'already lower');
+  });
+  it('should return "1,000,000." for "1,000,000.".toLower()', () => {
+    assert.equal('1,000,000.'.toLower(), '1,000,000.');
+  });
+});
+
+describe('Test for ucFirst method', () => {
+  it('should return FirSt for "firSt".ucFirst()', () => {
+    assert.equal('firSt'.ucFirst(), 'FirSt');
+  });
+  it('should return Andela "Andela".ucFirst()', () => {
+    assert.equal('Andela'.ucFirst(), 'Andela');
+  });
+  it('should return 12word "12word".ucFirst()', () => {
+    assert.equal('12word'.ucFirst(), '12word');
+  });
+});
+
+describe('Test for isQuestion method', () => {
+  it('should return true for "Is this a question?".isQuestion()', () => {
+    assert.equal('Is this a question?'.isQuestion(), true);
+  });
+  it('should return false for "Is this a ? question".isQuestion()', () => {
+    assert.equal('Is this a ? question'.isQuestion(), false);
+  });
+  it('should return true for "What?      ".isQuestion()', () => {
+    assert.equal('What?      '.isQuestion(), true);
+  });
+});
+
+describe('Test for words method', () => {
+  it('should return an array of the list of words ["It", "is", "working"]\
+    for "It is working".words()', () => {
+    assert.deepEqual('It is working'.words(), ['It', 'is', 'working']);
+  });
+  it('should return [] for "     ".words()', () => {
+    assert.deepEqual('     '.words(), []);
+  });
+  it('should return [I, have, $1,000,000,000] for \
+    I have $1,000,000,000 ".words()', () => {
+    assert.deepEqual('I have $1,000,000,000'.words(),
+      ['I', 'have', '1000000000']);
+  });
+});
+
+describe('Test for wordCount method', () => {
+  it('should return 3 for "It is working".wordCount()', () => {
+    assert.equal('It is working'.wordCount(), 3);
+  });
+  it('should return 5 for "Wierdly    spaced     text   \
+    contains 87938".wordCount()', () => {
+    assert.equal('Wierdly    spaced     text   contains 87938'.wordCount(),
+      5);
+  });
+});
+
+describe('Test for toCurrency method', () => {
+  it('should return 11,111.11 for 11111.11.toCurrency()', () => {
+    assert.equal('11111.11'.toCurrency(), '11,111.11');
+  });
+  it('should return 1,000,000 for 1000000.toCurrency()', () => {
+    assert.equal('1000000'.toCurrency(), '1,000,000');
```

Comment:
__Testing__
Numbers in currency format should always have trailing 2 decimal place values even for whole numbers. _i.e $1,000,000.00_
```JavaScript
 assert.equal('1000000'.toCurrency(), '1,000,000.00');
```